dissect/target/helpers/scrape.py

```diff
@@ -60,7 +60,14 @@ def find_needles(
             offset = fh.tell()
 
         read_size = min(block_size, end - offset) if end is not None else block_size
-        if not (next_block := fh.read(read_size)):
+        next_block = None
+
+        try:
+            next_block = fh.read(read_size)
```

Comment:
A normally behaving `fh.read()` shouldn't be able to raise `EOFError` but should return `b""` instead, can you explain or otherwise add a unit test for when this could happen?

Reply:
Sounds more like that is something that should be fixed 😄. Is there publicly available data or instructions on how to create reproducable data?

Reply:
Can you undo this change in the meantime?